lib/syslog/writer.go

```diff
@@ -244,6 +256,20 @@ func (c bufferedConn) Close() error                { return c.conn.Close() }
 func (c bufferedConn) Flush() error                { return c.buf.Flush() }
 func (c bufferedConn) Write(b []byte) (int, error) { return c.buf.Write(b) }
 
+func cachedDialWriter(network string, address string, config *tls.Config, socksProxy string) (w io.Writer, err error) {
```

Comment:
Is this function ever called concurrently?

Reply:
Possibly.  Should I put a mutex around it?

Reply:
Yeah, otherwise you may run into concurrent map access panics.  `sync.Map` was added to the stdlib recently, not sure if it's worth it over just making a mutex though.

Reply:
Yes, this code is accessed by multiple goroutines, proper synchronization needs to be put in place to manage the pool of connections.